### PR TITLE
Feature/sc 19280/upgrade clark to angular lts

### PR DESCRIPTION
**Pull request by shanioffe**
This is my first angular upgrade so please feel free to mark this up! I updated from v13 -> v14 then v14 -> v15.

### CODE REVIEW

**Review by pmanal1 — APPROVED** [review]
Good work!

File: .browserslistrc [hunk 1]
```diff
@@ -1,14 +0,0 @@
-# This file is used by the build system to adjust CSS and JS output to support the specified browsers below.
```
Comment (mfranz1) [inline .browserslistrc:1]: What was the reason for removing this file?

File: .husky/pre-commit [hunk 2]
```diff
@@ -1,4 +0,0 @@
-#!/bin/sh
```
Comment (mfranz1) [inline .husky/pre-commit:1]: This should not have been removed

File: package.json [hunk 3]
```diff
@@ -12,37 +12,36 @@
     "test:lighthouse:debug": "node --inspect node_modules/.bin/jest --watch -c ./lighthouse/jest.config.lighthouse.js --runInBand",
     "lint": "npx ng lint clark",
     "e2e": "ng e2e",
-    "storybook": "start-storybook",
-    "prepare": "husky install"
```
Comment (mfranz1) [inline package.json:16]: this prepare script should be added back

File: package.json [hunk 4]
```diff
@@ -12,37 +12,36 @@
...
+    "@angular/router": "^15.2.2",
     "@babel/core": "^7.16.7",
     "@cyber4all/clark-taxonomy": "^4.0.5",
     "aws-sdk": "^2.1058.0",
     "core-js": "^3.6.5",
     "md5": "^2.2.1",
     "ng-click-outside": "^9.0.1",
     "ng2-charts": "^2.3.0",
-    "ng2-ckeditor": "1.3.6",
+    "ng2-ckeditor": "1.3.7",
```
Comment (mfranz1) [inline package.json:42]: Was this tested? This can cause breaking changes. Please see the note dependencies note above.

File: package.json [hunk 5]
```diff
@@ -86,7 +85,7 @@
     "sass": "^1.46.0",
     "ts-jest": "^27.1.3",
     "tslib": "^1.10.0",
-    "typescript": "4.6.2"
+    "typescript": "^4.8.4"
```
Comment (mfranz1) [inline package.json:88]: This is a large upgrade to jump 2 minor patches on typescript. Everything compiles correctly?

File: src/app/admin/admin.module.ts [hunk 6]
```diff
@@ -28,8 +28,8 @@ import { PaginationComponent } from './components/pagination/pagination.componen
 import { RelevancyDateComponent } from './components/relevancy-date/relevancy-date.component';
 import { MatDatepickerModule } from '@angular/material/datepicker';
 import { MatNativeDateModule } from '@angular/material/core';
-import { MatFormFieldModule } from '@angular/material/form-field';
-import { MatInputModule } from '@angular/material/input';
+import { MatLegacyFormFieldModule as MatFormFieldModule } from '@angular/material/legacy-form-field';
```
Comment (mfranz1) [inline src/app/admin/admin.module.ts:31]: Could you provide the reference to the Angular documentation about this?

File: src/app/collection/collection.module.ts [hunk 7]
```diff
@@ -25,7 +25,7 @@ import { HeaderInfo502Component } from './pages/collection-502/components/header
 import { CuratorCardComponent } from './pages/collection-502/components/curators/components/curator-card/curator-card.component';
 import { TitleComponent } from './pages/collection-502/components/title/title.component';
 import { FeaturedComponent } from './pages/collection-502/components/featured/featured.component';
-import { MatSlideToggleModule } from '@angular/material/slide-toggle';
+import { MatLegacySlideToggleModule as MatSlideToggleModule } from '@angular/material/legacy-slide-toggle';
```
Comment (mfranz1) [inline src/app/collection/collection.module.ts:28]: Also would be interested in this documentation as well

File: src/app/components/subscription/subscription.component.html [hunk 8]
```diff
@@ -5,6 +5,6 @@
 <iframe
     class="clark-subscription"
     #newsletterSubscription
-    src="https://cdn.forms-content.sg-form.com/9a369384-51e4-11ec-a0b6-92cdb9380ecd"
     sandbox="allow-same-origin allow-scripts allow-forms"
+    src="https://cdn.forms-content.sg-form.com/9a369384-51e4-11ec-a0b6-92cdb9380ecd"
```
Comment (mfranz1) [inline src/app/components/subscription/subscription.component.html:9]: Just curious why was this moved? lol

File: src/app/cube/content-pages/about-us/about-us.component.html [hunk 9]
```diff
@@ -20,10 +20,12 @@ <h3>{{sec.title}}</h3>
             <div class="content">
               <div [innerHTML]="questions.answer"></div>
               <div *ngIf="questions.video">
-                <iframe src="https://www.youtube.com/embed/ymerRpfbnQQ" title="Clark Searching and Sorting Video"
+                <iframe title="Clark Searching and Sorting Video"
                   frameborder="0"
                   allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
-                  allowfullscreen>
+                  allowfullscreen
+                  src="https://www.youtube.com/embed/ymerRpfbnQQ"
```
Comment (mfranz1) [inline src/app/cube/content-pages/about-us/about-us.component.html:27]: Was this angular?

File: src/app/cube/cube.module.ts [hunk 10]
```diff
@@ -20,9 +20,9 @@ import { SharedModule } from '../shared/shared.module';
 import { CubeComponent } from './cube.component';
 import { BrowseComponent } from './browse/browse.component';
 import { UserProfileComponent } from './user-profile/user-profile.component';
-import { MatTabsModule } from '@angular/material/tabs';
+import { MatLegacyTabsModule as MatTabsModule } from '@angular/material/legacy-tabs';
```
Comment (mfranz1) [inline src/app/cube/cube.module.ts:23]: This legacy documenaton pls

File: src/assets/ckeditor/samples/toolbarconfigurator/lib/codemirror/codemirror.css [hunk 11]
```diff
@@ -64,17 +64,17 @@
   -moz-animation: blink 1.06s steps(1) infinite;
   animation: blink 1.06s steps(1) infinite;
 }
-@-moz-keyframes blink {
+@-moz-keyframes codemirror_blink {
```
Comment (mfranz1) [inline src/assets/ckeditor/samples/toolbarconfigurator/lib/codemirror/codemirror.css:67]: Was this part of the upgrade?

**Review by mfranz1 — CHANGES_REQUESTED** [review]
Good work! Please double check the client with it running to make sure everything still functions as intended. 
Also, The vars file I don't think should have been renamed, that could potentially cause breaking changes.

File: .husky/pre-commit (hunk 2 above)
Comment (Cwagne17) [reply .husky/pre-commit:1]: Agreeded, we should also upgrade the orb to use version 2.x.x

**Review by Cwagne17 — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (Maliyaa) [reply package.json:42]: +1

**Review by Maliyaa — COMMENTED** [review]
Great job Shannon! 😎 Just address Mike and Chris' comments

File: src/app/cube/content-pages/about-us/about-us.component.html (hunk 9 above)
Comment (pzalep1) [reply src/app/cube/content-pages/about-us/about-us.component.html:27]: This might have been linting. This page hasn't been touched in a while

File: .browserslistrc (hunk 1 above)
Comment (shanioffe) [reply .browserslistrc:1]: I'm actually not sure, I don't recall going in and deleting this file. Is it possible that it was automatically deleted when I ran ng update?

File: src/app/cube/content-pages/about-us/about-us.component.html (hunk 9 above)
Comment (shanioffe) [reply src/app/cube/content-pages/about-us/about-us.component.html:27]: This was me. I was following the recommendation here: https://angular.io/errors/NG0910

File: src/assets/ckeditor/samples/toolbarconfigurator/lib/codemirror/codemirror.css (hunk 11 above)
Comment (shanioffe) [reply src/assets/ckeditor/samples/toolbarconfigurator/lib/codemirror/codemirror.css:67]: Yes, https://angular.io/guide/update-to-version-15#v15-bc-03

File: .husky/pre-commit (hunk 2 above)
Comment (shanioffe) [reply .husky/pre-commit:1]: What version should we upgrade to?

File: src/app/admin/admin.module.ts (hunk 6 above)
Comment (shanioffe) [reply src/app/admin/admin.module.ts:31]: After some research i was able to find this explanation on [github](https://github.com/angular/components/blob/main/guides/v15-mdc-migration.md) (go to "What has changed?" section). 

My understanding is that these components have been deprecated, and can only be supported by the new version of angular if you use a legacy import.

File: src/app/collection/collection.module.ts (hunk 7 above)
Comment (shanioffe) [reply src/app/collection/collection.module.ts:28]: documentation: https://material.angular.io/guide/mdc-migration
See my comment under src/app/admin/admin.module.ts for more explanation

File: src/app/auth/forgot-password/forgot-password.component.ts [hunk 12]
```diff
@@ -1,5 +1,5 @@
 import { Component, OnDestroy, OnInit } from '@angular/core';
-import { FormGroup } from '@angular/forms';
+import { UntypedFormGroup } from '@angular/forms';
```
[the remark this answers is not in the document]
Comment (shanioffe) [reply src/app/auth/forgot-password/forgot-password.component.ts:2]: Based on the documentation [here](url), it looks like there were style changes, some appearances removed, and API changes made to the form-field component

File: src/app/collection/collection.module.ts (hunk 7 above)
Comment (shanioffe) [reply src/app/collection/collection.module.ts:28]: https://material.angular.io/guide/mdc-migration

File: src/app/cube/cube.module.ts (hunk 10 above)
Comment (shanioffe) [reply src/app/cube/cube.module.ts:23]: https://material.angular.io/guide/mdc-migration

File: src/app/auth/auth.module.ts [hunk 13]
```diff
@@ -8,7 +8,7 @@ import { ForgotPasswordComponent } from './forgot-password/forgot-password.compo
 import { EmailVerifiedComponent } from './email-verified/email-verified.component';
 import { RouterModule } from '@angular/router';
 import { AuthRoutingModule } from './auth.routing';
-import { MatInputModule } from '@angular/material/input';
+import { MatLegacyInputModule as MatInputModule } from '@angular/material/legacy-input';
```
[the remark this answers is not in the document]
Comment (shanioffe) [reply src/app/auth/auth.module.ts:11]: I just ran the migration command i found [here](https://material.angular.io/guide/mdc-migration), got rid of the legacy components!

File: package.json (hunk 4 above)
Comment (shanioffe) [reply package.json]: I downgraded back to v6, hope that helps.

File: package.json (hunk 5 above)
Comment (shanioffe) [reply package.json:89]: Good call, I downgraded to 4.7.4 and that helped. Still a work in progress.

**Review by Cwagne17 — APPROVED** [review]

**Review by jouypo23 — APPROVED** [review]

File: package.json [hunk 14]
```diff
@@ -20,32 +20,32 @@
...
+    "@angular/router": "^15.2.2",
     "@babel/core": "^7.16.7",
     "@cyber4all/clark-taxonomy": "^4.0.5",
     "aws-sdk": "^2.1058.0",
     "core-js": "^3.6.5",
     "md5": "^2.2.1",
     "ng-click-outside": "^9.0.1",
     "ng2-charts": "^2.3.0",
-    "ng2-ckeditor": "1.3.6",
+    "ng2-ckeditor": "^1.3.6",
```
Comment (mfranz1) [inline package.json:43]:
```suggestion
    "ng2-ckeditor": "1.3.6",
```
I would be curious to see if the package-lock upgraded this. This has given issues in the past as well.

File: package.json [hunk 15]
```diff
@@ -20,32 +20,32 @@
...
     "ngx-cookie": "^5.0.2",
-    "ngx-cookie-service": "^13.2.0",
+    "ngx-cookie-service": "^15.0.0",
     "ngx-virtual-scroller": "^4.0.3",
     "raven-js": "^3.22.3",
-    "request-promise": "^4.2.5",
+    "request-promise": "^0.0.1",
```
Comment (mfranz1) [inline package.json:48]: Do you recall what required this to be down graded?

File: src/app/collection/pages/collection-502/collection-502.component.scss [hunk 16]
```diff
@@ -29,11 +29,13 @@
     margin-left: 10px;
   }
 }
-::ng-deep .mat-slide-toggle.mat-checked .mat-slide-toggle-bar {
+/* TODO(mdc-migration): The following rule targets internal classes of slide-toggle that may no longer apply for the MDC version. */
```
Comment (mfranz1) [inline src/app/collection/pages/collection-502/collection-502.component.scss:32]: The documentation mentions we will have to verify these are still working.
`TODO(mdc-migration)`

File: .husky/pre-commit [hunk 17]
```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
-npm run lint && npm run test:unit
+npm run lint && npm run test:unit
```
Comment (mfranz1) [inline .husky/pre-commit:4]:
```suggestion
npm run lint && npm run test:unit

```
This should just remove it from commit history

**Review by mfranz1 — CHANGES_REQUESTED** [review]
I would like to help test a few things and address the TODOs of the upgrade before merging.

File: .browserslistrc (hunk 1 above)
Comment (shanioffe) [reply .browserslistrc:1]: After some research I couldn't find a good reason why this file may have been deleted, so I added it back in and ran `npx browserslist@latest --update-db` to make sure everything was up to date.
